packages/codec/lib/basic/decode/index.ts

```diff
@@ -36,6 +36,57 @@ export function* decodeBasic(
   debug("pointer %o", pointer);
 
   switch (dataType.typeClass) {
+    case "userDefinedValueType": {
+      const fullType = <Format.Types.UDVTType>(
+        Format.Types.fullType(dataType, info.userDefinedTypes)
+      );
+      if (!fullType.underlyingType) {
+        const error = {
+          kind: "UserDefinedTypeNotFoundError" as const,
+          type: fullType,
+        };
+        if (strict || options.allowRetry) {
```

Comment:
How does strict effect retry?

Reply:
Neither affects the other; they act fairly independently.  I'm guessing you meant to ask a more general question about why this condition?  I'll answer that instead if you don't mind. :)

In non-strict mode, when something goes wrong, we return an `ErrorResult`, whereas in strict mode, when something goes wrong, we throw.

Or rather, that's what we do for non-retriable errors; for retriable errors, if `allowRetry` is set, then what we do instead is to trigger a retry.  Of course, the way we trigger a retry is to throw an error, but set its `allowRetry` bit so that a retry will occur.

So, putting it together, on a retriable error, we throw if either `strict` *or* `allowRetry` is set, and we set its `allowRetry` flag based on the value of `options.allowRetry`.  Retries are for falling back from full mode to ABI mode, so this error, being unable to locate a user-defined type in full mode, is an example of a retriable error.  Thus, the condition.

Reply:
This indeed was the question I couldn't articulate. Thanks @haltman-at !

Reply:
Actually I just realized I made a slight mistake in the above -- for retriable errors, we don't set `allowRetry` on the error based on the `allowRetry` option; instead we just always set it to `true` regardless.  It's up to the caller to know whether it wants to retry or not.